src/shared/machine-image.c

```diff
@@ -221,6 +223,9 @@ static int image_make(
                 st = &stbuf;
         }
 
+        if (!path)
+                safe_getcwd(&cwd);
```

Comment:
This creates an unnecessary allocation in most cases. But it's a not a hot path, so we don't need to optimize.